packages/mobile-top-nav/src/media-menu.js

```diff
@@ -0,0 +1,153 @@
+import { LitElement, html } from 'lit-element';
+
+import './media-slider';
+import './assets/img/media-menu-images';
+import mediaMenuCss from './css/media-menu';
+
+const menuSelection = [
+  {
+    icon: 'web',
+    menu: 'web',
+    label: 'Wayback Machine',
+  },
+  {
+    icon: 'texts',
+    menu: 'texts',
+    label: 'Texts',
+  },
+  {
+    icon: 'video',
+    menu: 'video',
+    label: 'Video',
+  },
+  {
+    icon: 'audio',
+    menu: 'audio',
+    label: 'Audio',
+  },
+  {
+    icon: 'software',
+    menu: 'software',
+    label: 'Software',
+  },
+  {
+    icon: 'images',
+    menu: 'images',
+    label: 'Images',
+  },
+  {
+    icon: 'ellipses',
+    menu: 'more',
+    label: 'More',
+  },
+];
+
+class MediaMenu extends LitElement {
+  static get properties() {
+    return {
+      mediaMenuOpen: { type: Boolean },
+      mediaMenuAnimate: { type: Boolean },
+      mediaSliderOpen: { type: Boolean },
+      mediaSliderAnimate: { type: Boolean },
+      selectedMenuOption: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.mediaSliderOpen = false;
+    this.mediaSliderAnimate = false;
+    this.selectedMenuOption = '';
+  }
+
+  updated(changedProperties) {
+    const { mediaMenuOpen, mediaSliderOpen } = this;
+    const menuClosed = changedProperties.has('mediaMenuOpen')
+      && changedProperties.get('mediaMenuOpen')
+      && !mediaMenuOpen;
+
+    if (menuClosed && mediaSliderOpen) {
+      this.mediaSliderOpen = false;
+      this.mediaSliderAnimate = false;
+      this.selectedMenuOption = '';
+    }
+  }
+
+  closeMediaSlider() {
+    this.mediaSliderAnimate = true;
+    this.mediaSliderOpen = false;
+    this.selectedMenuOption = '';
+  }
+
+  toggleMediaSlider() {
+    if (!this.mediaSliderOpen) {
+      this.mediaSliderAnimate = true;
+      this.mediaSliderOpen = !this.mediaSliderOpen;
+    }
+  }
+
+  select(mediatype) {
+    const currentSelection = this.selectedMenuOption;
+
+    if (currentSelection === mediatype) {
+      this.closeMediaSlider();
+      return;
+    }
+
+    this.selectedMenuOption = mediatype;
+    this.toggleMediaSlider();
+  }
+
+  get mediaMenuOptionsTemplate() {
+    const buttons = menuSelection.map(({ icon, menu, label }) => {
+      const selected = this.selectedMenuOption === menu ? 'selected' : '';
+      return html`
+        <button class="menu-item ${selected}" @click="${this.select.bind(this, menu)}">
+          <span class="icon"
+            ><mediamenu-image
```

Comment:
Why are the closing brackets on the next line?

Reply:
I believe it's to swallow the whitespace characters that would be rendered and affect inline and inline-block elements.

Reply:
Is that a problem?